File: frontend/src/lib/cyberDomains.js
// Canonical list of cybersecurity investment domains.
// Keep in sync with backend/app/enrichment/cyber_domains.py (used in the AI prompt).

export const CYBER_DOMAINS = [
  // Application Security
  { key: 'sast', label: 'SAST', group: 'AppSec' },
  { key: 'dast', label: 'DAST', group: 'AppSec' },
  { key: 'iast', label: 'IAST', group: 'AppSec' },
  { key: 'sca', label: 'SCA', group: 'AppSec' },
  { key: 'aspm', label: 'ASPM', group: 'AppSec' },
  { key: 'api_security', label: 'API Security', group: 'AppSec' },
  { key: 'waap', label: 'WAF / WAAP', group: 'AppSec' },
  { key: 'secure_by_design', label: 'Secure by Design', group: 'AppSec' },
  { key: 'ai_pentesting', label: 'AI Pentesting', group: 'AppSec' },
  { key: 'pentesting', label: 'Pentesting / Red Team', group: 'AppSec' },
  { key: 'bug_bounty', label: 'Bug Bounty', group: 'AppSec' },

  // Cloud Security
  { key: 'cnapp', label: 'CNAPP', group: 'Cloud' },
  { key: 'cspm', label: 'CSPM', group: 'Cloud' },
  { key: 'cwpp', label: 'CWPP', group: 'Cloud' },
  { key: 'kspm', label: 'KSPM', group: 'Cloud' },
  { key: 'sspm', label: 'SSPM', group: 'Cloud' },
  { key: 'ciem', label: 'CIEM', group: 'Cloud' },
  { key: 'container_security', label: 'Container Security', group: 'Cloud' },
  { key: 'serverless_security', label: 'Serverless Security', group: 'Cloud' },

  // Exposure & Vulnerability Management
  { key: 'vulnerability_management', label: 'Vulnerability Mgmt', group: 'Exposure & VM' },
  { key: 'exposure_management', label: 'Exposure Mgmt', group: 'Exposure & VM' },
  { key: 'ctem', label: 'CTEM', group: 'Exposure & VM' },
  { key: 'asm_easm', label: 'ASM / EASM', group: 'Exposure & VM' },
  { key: 'caasm', label: 'CAASM', group: 'Exposure & VM' },
  { key: 'scanning', label: 'Scanning', group: 'Exposure & VM' },
  { key: 'prioritization', label: 'Prioritization', group: 'Exposure & VM' },
  { key: 'remediation', label: 'Remediation', group: 'Exposure & VM' },
  { key: 'patch_management', label: 'Patch Mgmt', group: 'Exposure & VM' },
  { key: 'posture_management', label: 'Posture Mgmt', group: 'Exposure & VM' },
  { key: 'bas', label: 'Breach & Attack Sim', group: 'Exposure & VM' },

  // Identity & Access
  { key: 'iam', label: 'IAM', group: 'Identity' },
  { key: 'iga', label: 'IGA', group: 'Identity' },
  { key: 'pam', label: 'PAM', group: 'Identity' },
  { key: 'itdr', label: 'ITDR', group: 'Identity' },
  { key: 'ciam', label: 'CIAM', group: 'Identity' },
  { key: 'secrets_management', label: 'Secrets Mgmt', group: 'Identity' },
  { key: 'mfa_passwordless', label: 'MFA / Passwordless', group: 'Identity' },

  // Endpoint
  { key: 'edr', label: 'EDR', group: 'Endpoint' },
  { key: 'xdr', label: 'XDR', group: 'Endpoint' },
  { key: 'epp', label: 'EPP', group: 'Endpoint' },
  { key: 'mdm_uem', label: 'MDM / UEM', group: 'Endpoint' },
  { key: 'mobile_security', label: 'Mobile Security', group: 'Endpoint' },

  // Network
  { key: 'ndr', label: 'NDR', group: 'Network' },
  { key: 'firewall_ngfw', label: 'Firewall / NGFW', group: 'Network' },
  { key: 'ztna', label: 'ZTNA', group: 'Network' },
  { key: 'sase', label: 'SASE', group: 'Network' },
  { key: 'sse', label: 'SSE', group: 'Network' },
  { key: 'swg', label: 'SWG', group: 'Network' },
  { key: 'microsegmentation', label: 'Microsegmentation', group: 'Network' },
  { key: 'ddos_protection', label: 'DDoS Protection', group: 'Network' },

  // Data
  { key: 'dlp', label: 'DLP', group: 'Data' },
  { key: 'dspm', label: 'DSPM', group: 'Data' },
  { key: 'data_encryption', label: 'Data Encryption', group: 'Data' },
  { key: 'database_security', label: 'Database Security', group: 'Data' },

  // Detection & Response
  { key: 'siem', label: 'SIEM', group: 'Detection & Response' },
  { key: 'soar', label: 'SOAR', group: 'Detection & Response' },
  { key: 'mdr', label: 'MDR', group: 'Detection & Response' },
  { key: 'threat_intelligence', label: 'Threat Intel', group: 'Detection & Response' },
  { key: 'deception', label: 'Deception', group: 'Detection & Response' },
  { key: 'ueba', label: 'UEBA', group: 'Detection & Response' },

  // GRC & Risk
  { key: 'grc', label: 'GRC', group: 'GRC & Risk' },
  { key: 'compliance_automation', label: 'Compliance Automation', group: 'GRC & Risk' },
  { key: 'tprm', label: 'Third-Party Risk', group: 'GRC & Risk' },
  { key: 'privacy', label: 'Privacy', group: 'GRC & Risk' },
  { key: 'insider_risk', label: 'Insider Risk', group: 'GRC & Risk' },

  // Email & Web
  { key: 'email_security', label: 'Email Security', group: 'Email & Web' },
  { key: 'anti_phishing', label: 'Anti-Phishing / BEC', group: 'Email & Web' },
  { key: 'browser_security', label: 'Browser Security', group: 'Email & Web' },
  { key: 'bot_management', label: 'Bot Management', group: 'Email & Web' },
  { key: 'fraud_prevention', label: 'Fraud Prevention', group: 'Email & Web' },

  // OT / IoT
  { key: 'ot_security', label: 'OT / ICS Security', group: 'OT & IoT' },
  { key: 'iot_security', label: 'IoT Security', group: 'OT & IoT' },

  // AI Security
  { key: 'ai_spm', label: 'AI-SPM', group: 'AI Security' },
  { key: 'llm_security', label: 'LLM / GenAI Security', group: 'AI Security' },
  { key: 'model_security', label: 'Model Security', group: 'AI Security' },

  // Software Supply Chain
  { key: 'supply_chain_security', label: 'Supply Chain Security', group: 'Supply Chain' },
  { key: 'sbom', label: 'SBOM', group: 'Supply Chain' },
  { key: 'iac_security', label: 'IaC Security', group: 'Supply Chain' },
  { key: 'secrets_scanning', label: 'Secrets Scanning', group: 'Supply Chain' },
  { key: 'code_security', label: 'Code Security', group: 'Supply Chain' },

  // Other
  { key: 'security_awareness', label: 'Security Awareness', group: 'Other' },
  { key: 'quantum_pqc', label: 'Quantum / PQC', group: 'Other' },
  { key: 'hardware_security', label: 'Hardware Security', group: 'Other' },
  { key: 'devsecops', label: 'DevSecOps', group: 'Other' },
]

export const CYBER_DOMAIN_KEYS = CYBER_DOMAINS.map((d) => d.key)

export const CYBER_DOMAIN_GROUPS = CYBER_DOMAINS.reduce((acc, d) => {
  if (!acc[d.group]) acc[d.group] = []
  acc[d.group].push(d)
  return acc
}, {})

File: backend/app/enrichment/cyber_domains.py
"""Canonical cybersecurity investment domains.

Keep in sync with frontend/src/lib/cyberDomains.js — both files enumerate the
landscape used to map VC portfolio coverage. The AI enrichment prompt instructs
Vertex to return a boolean for every key here.
"""
from __future__ import annotations

CYBER_DOMAINS: list[tuple[str, str, str]] = [
    # (key, label, group)
    ("sast", "SAST", "AppSec"),
    ("dast", "DAST", "AppSec"),
    ("iast", "IAST", "AppSec"),
    ("sca", "SCA", "AppSec"),
    ("aspm", "ASPM", "AppSec"),
    ("api_security", "API Security", "AppSec"),
    ("waap", "WAF / WAAP", "AppSec"),
    ("secure_by_design", "Secure by Design", "AppSec"),
    ("ai_pentesting", "AI Pentesting", "AppSec"),
    ("pentesting", "Pentesting / Red Team", "AppSec"),
    ("bug_bounty", "Bug Bounty", "AppSec"),

    ("cnapp", "CNAPP", "Cloud"),
    ("cspm", "CSPM", "Cloud"),
    ("cwpp", "CWPP", "Cloud"),
    ("kspm", "KSPM", "Cloud"),
    ("sspm", "SSPM", "Cloud"),
    ("ciem", "CIEM", "Cloud"),
    ("container_security", "Container Security", "Cloud"),
    ("serverless_security", "Serverless Security", "Cloud"),

    ("vulnerability_management", "Vulnerability Management", "Exposure & VM"),
    ("exposure_management", "Exposure Management", "Exposure & VM"),
    ("ctem", "CTEM", "Exposure & VM"),
    ("asm_easm", "ASM / EASM", "Exposure & VM"),
    ("caasm", "CAASM", "Exposure & VM"),
    ("scanning", "Scanning", "Exposure & VM"),
    ("prioritization", "Prioritization", "Exposure & VM"),
    ("remediation", "Remediation", "Exposure & VM"),
    ("patch_management", "Patch Management", "Exposure & VM"),
    ("posture_management", "Posture Management", "Exposure & VM"),
    ("bas", "Breach & Attack Simulation", "Exposure & VM"),

    ("iam", "IAM", "Identity"),
    ("iga", "IGA", "Identity"),
    ("pam", "PAM", "Identity"),
    ("itdr", "ITDR", "Identity"),
    ("ciam", "CIAM", "Identity"),
    ("secrets_management", "Secrets Management", "Identity"),
    ("mfa_passwordless", "MFA / Passwordless", "Identity"),

    ("edr", "EDR", "Endpoint"),
    ("xdr", "XDR", "Endpoint"),
    ("epp", "EPP", "Endpoint"),
    ("mdm_uem", "MDM / UEM", "Endpoint"),
    ("mobile_security", "Mobile Security", "Endpoint"),

    ("ndr", "NDR", "Network"),
    ("firewall_ngfw", "Firewall / NGFW", "Network"),
    ("ztna", "ZTNA", "Network"),
    ("sase", "SASE", "Network"),
    ("sse", "SSE", "Network"),
    ("swg", "SWG", "Network"),
    ("microsegmentation", "Microsegmentation", "Network"),
    ("ddos_protection", "DDoS Protection", "Network"),

    ("dlp", "DLP", "Data"),
    ("dspm", "DSPM", "Data"),
    ("data_encryption", "Data Encryption", "Data"),
    ("database_security", "Database Security", "Data"),

    ("siem", "SIEM", "Detection & Response"),
    ("soar", "SOAR", "Detection & Response"),
    ("mdr", "MDR", "Detection & Response"),
    ("threat_intelligence", "Threat Intelligence", "Detection & Response"),
    ("deception", "Deception", "Detection & Response"),
    ("ueba", "UEBA", "Detection & Response"),

    ("grc", "GRC", "GRC & Risk"),
    ("compliance_automation", "Compliance Automation", "GRC & Risk"),
    ("tprm", "Third-Party Risk Management", "GRC & Risk"),
    ("privacy", "Privacy", "GRC & Risk"),
    ("insider_risk", "Insider Risk", "GRC & Risk"),

    ("email_security", "Email Security", "Email & Web"),
    ("anti_phishing", "Anti-Phishing / BEC", "Email & Web"),
    ("browser_security", "Browser Security", "Email & Web"),
    ("bot_management", "Bot Management", "Email & Web"),
    ("fraud_prevention", "Fraud Prevention", "Email & Web"),

    ("ot_security", "OT / ICS Security", "OT & IoT"),
    ("iot_security", "IoT Security", "OT & IoT"),

    ("ai_spm", "AI-SPM", "AI Security"),
    ("llm_security", "LLM / GenAI Security", "AI Security"),
    ("model_security", "Model Security", "AI Security"),

    ("supply_chain_security", "Supply Chain Security", "Supply Chain"),
    ("sbom", "SBOM", "Supply Chain"),
    ("iac_security", "IaC Security", "Supply Chain"),
    ("secrets_scanning", "Secrets Scanning", "Supply Chain"),
    ("code_security", "Code Security", "Supply Chain"),

    ("security_awareness", "Security Awareness", "Other"),
    ("quantum_pqc", "Quantum / PQC", "Other"),
    ("hardware_security", "Hardware Security", "Other"),
    ("devsecops", "DevSecOps", "Other"),
]

CYBER_DOMAIN_KEYS: list[str] = [k for k, _, _ in CYBER_DOMAINS]


def render_domains_for_prompt() -> str:
    """Render the canonical domain list as grouped bullets for the AI prompt."""
    by_group: dict[str, list[tuple[str, str]]] = {}
    for key, label, group in CYBER_DOMAINS:
        by_group.setdefault(group, []).append((key, label))

    lines: list[str] = []
    for group, items in by_group.items():
        lines.append(f"  {group}:")
        for key, label in items:
            lines.append(f"    - {key}  ({label})")
    return "\n".join(lines)
